evm/src/main/java/org/hyperledger/besu/evm/EVM.java

```diff
@@ -277,6 +275,18 @@ public void runToHalt(final MessageFrame frame, final OperationTracer tracing) {
           case 0x9f:
             result = SwapOperation.staticOperation(frame, opcode - SWAP_BASE);
             break;
+          case 0xb0: // CALLF
+          case 0xb1: // RETF
+          case 0xb2: // JUMPF
+            // Function operations reset code
+            if (enableShanghai) {
```

Comment:
I've been thinking about how to implement this for #4760 and I have some doubts about this mechanism. Do you think this will be enough in a not so distant future when/if `JUMP`/`JUMPI` gets deprecated (or `PC`)? Will we have only two sets of operations, let's say _legacy_ and _oef_, "forever"? Probably we can use it like this so far, but maybe we could revisit this after Shangai

Reply:
Two loops may have positive performance impacts.  However it also introduces risk if there are cross-version opcode changes.

The advantage of EOF is we can have one "catch all" loop for all of EOF and rely on code validation from permitting bad opcodes from being marked as valid EVM code.